eglot-jl.el

```diff
@@ -83,9 +68,8 @@ Otherwise returns nil"
   "Return list of strings to be called to start the Julia language server."
   `(,eglot-jl-julia-command
     ,@eglot-jl-julia-flags
-    ,(concat "--project=" eglot-jl-base)
+    "--project"
```

Comment:
Is the ls-invocation guaranteed to be called in the same directory as the file the language server is being invoked for? If so is it a documented part of eglot or an implementation detail?

Reply:
Ah, this is an issue...

Looking at [eglot.el](https://github.com/joaotavora/eglot/blob/da7ff4883d3108468265fdc16a217d8e6aff92f2/eglot.el#L849), it looks like the server is called from the root directory of the project. Although I did not find it clearly documented, this looks like it is part of the API (since projects are so central in how the buffer/server association is managed)

I think that definitely rules out the `--project` hack. I'm going to pass the current source directory as argument to `eglot-jl.jl`, and re-use the internal Julia logic to compute the project directory from there (see below).